salt/states/heat.py

```diff
@@ -0,0 +1,342 @@
+                kwargs=None)
+
+            template_manage_result = __salt__['file.manage_file'](
+                name=template_tmp_file,
+                sfn=tsfn,
+                ret=None,
+                source=template,
+                source_sum=source_sum,
+                user=None,
+                group=None,
+                mode=None,
+                saltenv='base',
+                backup=None,
+                makedirs=True,
+                template=None,
+                show_changes=False,
+                contents=None,
+                dir_mode=None)
+
+            if (template_manage_result['result']) or \
+                    ((__opts__['test']) and (template_manage_result['result'] is not False)):
+                with salt.utils.fopen(template_tmp_file, 'r') as tfp_:
+                    tpl = tfp_.read()
+                    salt.utils.safe_rm(template_tmp_file)
+                    try:
+                        if isinstance(tpl, six.binary_type):
+                            tpl = tpl.decode('utf-8')
+                        template_parse = _parse_template(tpl)
+                        if 'heat_template_version' in template_parse:
+                            template_new = yaml.dump(template_parse, Dumper=YamlDumper)
+                        else:
+                            template_new = jsonutils.dumps(template_parse, indent=2, ensure_ascii=False)
+                        salt.utils.safe_rm(template_tmp_file)
+                    except ValueError as ex:
+                        ret['result'] = False
+                        ret['comment'] = 'Error parsing template {0}'.format(ex)
+            else:
+                ret['result'] = False
+                ret['comment'] = "Can't open template: {0} {1}".format(template, comment_)
+        else:
+            ret['result'] = False
+            ret['comment'] = "Can't open template"
+        if ret['result'] is True:
+            template_stack = __salt__['heat.template_stack'](name=name, profile=profile)
+            if not template_stack['result']:
+                ret['result'] = False
+                ret['comment'] = template_stack['comment']
+        if ret['result'] is False:
+            return ret
+
+        try:
+            checksum_template = __salt__['hashutil.digest'](template_new)
+            checksum_stack = __salt__['hashutil.digest'](template_stack['template'])
+        except salt.exceptions.CommandExecutionError as cmdexc:
+            ret['result'] = False
+            ret['comment'] = '{0}'.format(cmdexc)
+
+        if ret['result'] is True:
+            if checksum_template == checksum_stack:
+                if __opts__['test']:
+                    ret['result'] = True
+                    ret['comment'] = 'Stack {0} is deployed'.format(name)
+                    return ret
+                else:
+                    ret['result'] = False
+                    ret['comment'] = 'Template ha same checksum: {0} {1}'\
```

Comment:
Typing error (Italian avere), 'has' is correct. Maybe better 'Templates have same checksum: {0} {1}'..... .